src/entities/addLiquidityBoosted/doAddLiquidityPropotionalQuery.ts

```diff
@@ -0,0 +1,31 @@
+import { createPublicClient, http, getAddress } from 'viem';
+
+import { BALANCER_COMPOSITE_LIQUIDITY_ROUTER, CHAINS } from '@/utils';
+
+import { AddLiquidityProportionalInput } from '../addLiquidity/types';
+import { Address } from '@/types';
+
+import { balancerCompositeLiquidityRouterAbi } from '@/abi';
+
+export const doAddLiquidityProportionalQuery = async (
+    { rpcUrl, chainId, referenceAmount }: AddLiquidityProportionalInput,
```

Comment:
I would change this to accept user address and userData. (Not as optionals, index.ts can handle defaults)

Reply:
I have addressed this as part of a larger refactor. It now allows to optionally supply `userData` & `userAddress` when querying the add/Remove classes and enforces values to be provided in the queries. Let me know what you think of that approach. It makes typing a bit overlapping as the same properties are either optionally or required.